addons/pinniped/config-controller/controllers/handlers.go

```diff
@@ -100,3 +94,43 @@ func (c *PinnipedV3Controller) withPackageName(packageName string) builder.Predi
 		},
 	)
 }
+
+func (c *PinnipedV1Controller) addonSecretToCluster(o client.Object) []ctrl.Request {
```

Comment:
Does this mean `addon secret FOR cluster`?

Reply:
In a way, yes, we are mapping the addon secret to the cluster it belongs to and sending the cluster name/namespace to `Reconcile`.  The convention for these MapFunc seems to be `somethingToSomething` (check out `addon_controller.go`, where we have `BOMConfigMapToClusters` and `KubeadmControlPlaneToClusters`).  I'll try to find some docs on this.

Reply:
Cool cool.